app/Template/user_list/user_details.php

```diff
@@ -11,7 +11,10 @@
         <span><a href="mailto:<?= $this->text->e($user['email']) ?>"><?= $this->text->e($user['email']) ?></a></span>
     <?php endif ?>
 
-    <?php if (! empty($this->user->getGroupNames($user['id'])) ): ?>
-        <span><i class="fa fa-fw fa-group aria-hidden="true"></i> <?= $this->user->getGroupNames($user['id']) ?></span>
+    <?php if ( SHOW_GROUP_MEMBERSHIPS_IN_USERLIST ): ?>
+        <?php $groups_list_tooltip = t('%s is a member of the following group(s):', $user['name']) . '&#10;' . $this->user->getUsersGroupNames($user['id'])['full_list']; ?>
+        <?php if ($this->user->getUsersGroupNames($user['id'])['has_groups']): ?>
+            <span><i class="fa fa-fw fa-group aria-hidden="true" title="<?= $groups_list_tooltip ?>"></i> <?= $this->user->getUsersGroupNames($user['id'])['limited_list'] ?></span>
```

Comment:
When you are doing this `$this->user->getUsersGroupNames($user['id'])['limited_list']`, PHP will call your function first and then return the result. That means you are still executing 3 queries. You should store the result in a variable to avoid this.

For example:

```php
<?php $userGroups = $this->user->getUsersGroupNames($user['id']); ?>
```

And then you just use your array: `$userGroups['limited_list']`.

Reply:
I am a little embarrassed, because I was so focused on reducing the 3 methods in the helper to just 1, that I forgot about reducing the actual method-calls :-/

I guess that's what code-reviews are good for 👍 

Thanks for your help ... hope I solved all issues now.